### PR TITLE
[NPU] fix some npu error with OffloaderV2

### PR DESCRIPTION
**Pull request by zhangshuren416**
## Motivation
 when set --offload-mode=meta or sharded_gpu in feature offloaderV2, it doesn't work with npu.Then I fix some error to support those features in npu.

## Modifications
1.fix attr weight_loader of params missing in func `_move_param_to_meta`
2.add npu class `torch_npu.multiprocessing.reductions.` to `SafeUnpickler.ALLOWED_MODULE_PREFIXES`

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/utils/offloader.py [hunk 1]
```diff
@@ -452,6 +452,9 @@ def _move_param_to_meta(module, param_name):
             data=new_data,
             requires_grad=False,
         )
+        empty_weight_loader = lambda *args, **kwargs: None
+        weight_loader = getattr(old_param, "weight_loader", empty_weight_loader)
+        setattr(new_param, "weight_loader", weight_loader)
```
Comment (ping1jing2) [inline python/sglang/srt/utils/offloader.py:457]: Hi, thank you for working on it, and could you please add a ut for this case？

Comment (ping1jing2) [inline python/sglang/srt/utils/offloader.py:457]: never use `getattr/setattr` in sglang
